### DISCUSSION TITLE
I am getting Could not load type 'Elsa.Models.LogEntry' from assembly 'Elsa.Abstractions, Version=2.14.1.0, Culture=neutral, PublicKeyToken=null'.

### DISCUSSION THREAD

**Discussion by kajasumanie** [elsa-workflows/elsa-core#5414] [Q&A]
I've written a simple controller class to read data, but I'm encountering the following issue. Has anyone else faced the same problem?

I am getting Could not load type 'Elsa.Models.LogEntry' from assembly 'Elsa.Abstractions, Version=2.14.1.0, Culture=neutral, PublicKeyToken=null'.

        [HttpPost("v1/bins")]
        public async Task<IActionResult> Post([FromBody] BinRequestModel request, CancellationToken cancellationToken)
        {

            using var scope = _serviceProvider.CreateScope();
            var dbContextFactory = scope.ServiceProvider.GetRequiredService<IDbContextFactory<ElsaContext>>();

            using var dbContext = dbContextFactory.CreateDbContext();

            var data =  dbContext.WorkflowInstances.FirstOrDefault();

            return Ok("<h1>Request for Approval Sent</h1><p>Your bin has been received and will be reviewed shortly.</p>");
        }

**Comment by glime-ai** [comment]
To address the issue you're encountering with the `Could not load type 'Elsa.Models.LogEntry' from assembly 'Elsa.Abstractions, Version=2.14.1.0, Culture=neutral, PublicKeyToken=null'`, it's important to ensure that you have the correct version of the Elsa packages installed and that there are no version conflicts between different Elsa packages.

Here are a few steps you can take to troubleshoot and resolve the issue:

1. **Check for Package Versions**: Ensure that all Elsa-related NuGet packages are of the same version. Mismatched versions can lead to type load exceptions.

2. **Restore Packages**: Run a package restore to ensure all dependencies are correctly installed. You can do this via the NuGet Package Manager or by running `dotnet restore` in the terminal.

3. **Clean and Rebuild**: Sometimes, cleaning the solution and rebuilding it can resolve these types of issues. In Visual Studio, you can use the "Clean Solution" and "Rebuild Solution" options.

4. **Check for Breaking Changes**: If you recently upgraded Elsa, check the release notes for any breaking changes that might require changes in your code.

5. **Check the Project File**: Ensure that the project file (.csproj) references the correct version of the Elsa packages.

6. **Check Compatibility**: If you are using .NET Core or .NET 5+, make sure that the Elsa version you are using is compatible with your project's target framework.

If you continue to face issues after trying the above steps, consider reaching out to the Elsa community or checking the Elsa GitHub repository for similar issues or to open a new issue.

Since you mentioned a specific version of Elsa (`Version=2.14.1.0`), it's also possible that there might be a specific issue with that version, so looking for version-specific information in the Elsa documentation or release notes might be helpful.

Unfortunately, without more context or the ability to see the full project setup, it's difficult to provide a more precise solution. If you can provide additional details or clarify the context, I may be able to offer more targeted advice.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
